src/test/scala/DateParserAppSuite.scala

```diff
@@ -0,0 +1,32 @@
+import org.scalatest.freespec.AnyFreeSpec
+import wvlet.log.{LogLevel, Logger}
+
+import java.text.ParseException
+
+class DateParserAppSuite extends AnyFreeSpec {
+  Logger.setDefaultLogLevel(LogLevel.DEBUG)
+  val date_parser = new DateParserApp()
+  "DateParserAppSuite" - {
+    "date_parser" - {
+      "should parse good date with good format " in {
+        assert(date_parser.parse_date("yyyy-MM-dd", "2020-12-12").toString == "Sat Dec 12 00:00:00 UTC 2020")
+      }
+      "should raise exceptions and log messages for bad inputs" in {
+        assertThrows[ParseException](date_parser.parse_date("yyyy-MM-dd", "aaag-12-12-12"))
+        assertThrows[IllegalArgumentException](date_parser.parse_date("ggg-ggg-ggg", "aaag-12-12-12"))
+        // TODO how to test log messages itself?
```

Comment:
Any ideas?

Reply:
We don't test log messages (e.g. messages produced by `log.info()` or similar), but we test messages generated by exceptions, like this:
```scala
    val ex = intercept[IllegalArgumentException] {
      doSomethingBad()
    }
    assert(ex.getMessage.contains("A part of the expected error message"))
```
We usually don't check the full message because the purpose here is that the expected exception was thrown on the expected behavior. Sometimes, if a message is complicated, we test message generation logic by comparing the full message of the exception.